### DISCUSSION TITLE
Firebase messaging plugin update

### DISCUSSION THREAD

**Discussion by alejandroaap** [firebase/flutterfire#3362] [General] [answered]
How is rewriting the firebase messaging plugin going? Will there be the option to handle notifications with data when the application is finished?

**Comment by Ehesp** [comment, marked as answer]
Not started yet. Although will be looked at in the next few weeks. We're still looking into functionality of what's possible & can be improved. We'll update the https://github.com/FirebaseExtended/flutterfire/issues/2582 once we can give more updates.
